core/src/main/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingService.scala

```diff
@@ -133,6 +145,90 @@ object SpendReportingService {
+    allRows: List[FieldValueList],
+    start: DateTime,
+    end: DateTime,
+    names: Map[GoogleProjectId, WorkspaceName]
+  ): SpendReportingResults = {
+
+    var total = BigDecimal(0.0)
+    var total_credits = BigDecimal(0.0)
+
+    val currency = allRows.map(_.get("currency").getStringValue).distinct match {
+      case head :: _ => Currency.getInstance(head)
+      case _         => throw RawlsExceptionWithErrorReport(StatusCodes.NotFound, "No currencies found for spend data")
+    }
+
+    val all = allRows.map { row =>
+      val currencyString = row.get("currency").getStringValue
+      val currencyCode = Currency.getInstance(currencyString)
+      val projectId = row.get("project_id").getStringValue
+      val workspaceName = names.getOrElse(
+        GoogleProjectId(projectId),
+        throw RawlsExceptionWithErrorReport(
+          StatusCodes.InternalServerError,
+          s"unexpected project $projectId returned by BigQuery"
+        )
+      )
+
+      def getRoundedNumericValue(field: String): BigDecimal =
+        BigDecimal(row.get(field).getDoubleValue)
+          .setScale(currencyCode.getDefaultFractionDigits, RoundingMode.HALF_EVEN)
+
+      val subAggregation = List(
+        SpendReportingForDateRange(
+          getRoundedNumericValue("other_cost").toString,
+          getRoundedNumericValue("credits").toString,
+          currencyCode.toString,
+          Option(start),
+          Option(end),
+          category = Option(TerraSpendCategories.Other)
+        ),
+        SpendReportingForDateRange(
+          getRoundedNumericValue("storage_cost").toString,
+          getRoundedNumericValue("credits").toString,
+          currencyCode.toString,
+          category = Option(TerraSpendCategories.Storage)
+        ),
+        SpendReportingForDateRange(
+          getRoundedNumericValue("compute_cost").toString,
+          getRoundedNumericValue("credits").toString,
+          currencyCode.toString,
+          category = Option(TerraSpendCategories.Compute)
+        )
+      )
+
+      val total_cost = getRoundedNumericValue("total_cost")
+      val credits = getRoundedNumericValue("credits")
+      total = total + total_cost
+      total_credits = total_credits + credits
+
+      val workspaceTotal = SpendReportingForDateRange(
+        total_cost.toString,
+        credits.toString,
+        currencyCode.toString,
+        Option(start),
+        Option(end),
+        workspace = Option(workspaceName),
+        googleProjectId = Option(GoogleProject(projectId)),
+        subAggregation = Option(SpendReportingAggregation(SpendReportingAggregationKeys.Category, subAggregation))
+      )
+
+      SpendReportingAggregation(SpendReportingAggregationKeys.Workspace, List(workspaceTotal))
+
+    }
+
+    val summary = SpendReportingForDateRange(
+      total.toString,
+      total_credits.toString,
+      currency.toString, // TODO: what to do about combined summary for currencies?
```

Comment:
I have the same question! Maybe for this version of the report we need to throw an error if we find multiple currencies.

Reply:
I feel like it shouldn't throw an error - for a single billing project it should be a single currency, but since this covers multiple billing projects it seems plausible that there might be different currencies.  So I think ideally we would either not have a summary, or have a summary per currency?  Not really sure what the best thing to do here is but I don't think it should be error throwing.  We could potentially do that for a first pass and deal with it later on?